### PR TITLE
Add posture type to rule->benchmark

### PR DESCRIPTION
**Pull request by ari-aviran**
### Summary of your changes
Add posture type field to rule.benchmark

### Related Issues
https://github.com/elastic/security-team/issues/5732

### CODE REVIEW

File: dev/generate_rule_metadata.py [hunk 1]
```diff
@@ -80,6 +83,11 @@ class Rule:
     }
 }
 
+benchmark_to_posture_type = {
```
Comment (uri-weisman) [inline dev/generate_rule_metadata.py:86]: Maybe I'm not updated but I thought we decided to use the `posture_type` propagated as part of the agent config...
Will save us to maintain such maps.

Comment (ari-aviran) [reply dev/generate_rule_metadata.py:86]: I thought about it and this feels more correct to me. `cloudbeat` doesn't touch the `rule` object in the findings at all. @oren-zohar WDYT?

Comment (uri-weisman) [reply dev/generate_rule_metadata.py:86]: That's true for now, but I can think of use in such a field.
For example, determining if a leader election mechanism should be active.

Comment (oren-zohar) [reply dev/generate_rule_metadata.py:86]: I agree with @ari-aviran, cloudbeat enriches the events, but never touches the `rule` part

**Review by oren-zohar — CHANGES_REQUESTED** [review]
I think we need to add validation to `bundle/compliance/lib/output_validations/output_validations.rego`

**Review by oren-zohar — APPROVED** [review]
make sure to also update https://github.com/elastic/cloudbeat/blob/c0a256eb632f734056811cafac7fc6007542636b/evaluator/evaluator.go#L78

**Pull request merged**